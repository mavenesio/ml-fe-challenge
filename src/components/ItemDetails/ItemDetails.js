import React from 'react';
import styled from 'styled-components';
import Gallery from '../Gallery/Gallery';

const ItemDetailsContainer = styled.div`
    display:flex;
    flex-direction:column;
    justify-content:center;
    background-color:white;
    border-radius:4px;
`;
const ItemBox = styled.div`
    display:flex;
    flex-direction:row;
    justify-content:space-between;
    margin-top:16px;
`;
const ItemDetailsDescription = styled.div`
    display:flex;
    flex-direction:column;
    justify-content:space-between;
    padding-top: 4rem;
    margin-bottom:16px;
`;

const ItemImageContainer = styled.div`
    width:1000px;
    height:500px;
`;
const ItemDetailsColumn = styled.div`
    display:flex;
    flex-direction:column;
    justify-content:flex-start;
    padding:1rem;
`;
const ItemSubTitle = styled.div`
    align-self:flex-start;
    margin-top:32px;
    font-size:14px;
    font-family: Arial, Helvetica, sans-serif;

`;
const ItemTitle = styled.div`
    align-self:flex-start;
    margin-top:16px;
    font-size:24px;
    font-family: Arial, Helvetica, sans-serif;

`;
const ItemPrice = styled.div`
    align-self:flex-start;
    margin-top:32px;
    font-size:46px;
    font-family: Arial, Helvetica, sans-serif;
`;
const DescriptionTitle = styled.div`
    margin: 0px 0px 32px 32px;
    font-size:28px;
    font-family: Arial, Helvetica, sans-serif;
`;
const DescriptionText = styled.div`
    margin-bottom: 32px;
    padding: 16px;
    font-size:16px;
    font-family: Arial, Helvetica, sans-serif;
`;
const CustomButton = styled.button`
    background-color:#3483FA;
    color:white;
    border-radius:4px;
    margin-top: 32px;
    font-size:16px;
    height: 50px;
    width:100%;
    font-family: Arial, Helvetica, sans-serif;
    border:unset;
    cursor: pointer;
    &:focus {
        outline: unset;
    }
`;


const ItemDetails = ({details}) => {

    return (
        <div>
            {
                details && 
                <ItemDetailsContainer>
                    <ItemBox>
                        <ItemImageContainer>
                            <Gallery pictures={details.pictures}/>
                        </ItemImageContainer>
                        <ItemDetailsColumn>
                            <ItemSubTitle>{details.condition} - {`${details.sold_quantity} vendido${details.sold_quantity > 1 ? 's' : ''}`}</ItemSubTitle>
                            <ItemTitle>{details.title}</ItemTitle>
                            <ItemPrice>
                            {`${details.price.currency} ${details.price.amount.toFixed().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ".")} `}    
                            </ItemPrice>
                            <CustomButton onClick={()=> alert("This is not more available!")}>Comprar</CustomButton>
                        </ItemDetailsColumn>
                    </ItemBox>
                    <ItemDetailsDescription>
                        <DescriptionTitle>Descripcion del producto</DescriptionTitle>
                        <DescriptionText>{details.description}</DescriptionText>
                    </ItemDetailsDescription>
                </ItemDetailsContainer>
            }
        </div>
    )
}

export default ItemDetails;