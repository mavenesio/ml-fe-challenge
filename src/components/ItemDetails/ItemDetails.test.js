import React from 'react';
import ItemDetails from './ItemDetails';

describe('ItemDetails', () => {
  it('should render without props', () => {   
    expect(<ItemDetails />).toMatchSnapshot();
  });

  it('should render with props', () => {
    const details = {
        id: "MLA846526085",
        title: "Apple iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram",
        price:{
            currency: "$",
            amount: 139989.99,
            decimals: 0
        },
        picture: "http://mla-s2-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg",
        condition: "new",
        free_shipping: false,
        sold_quaranty: 1,
        description: "Mayor rendimiento↵Su memoria RAM de 3 GB te permitirá ej..",
    }  
    expect(<ItemDetails details={details} />).toMatchSnapshot();
  });
});
