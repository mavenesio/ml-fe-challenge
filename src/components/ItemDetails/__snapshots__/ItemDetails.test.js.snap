// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails should render with props 1`] = `
<ItemDetails
  details={
    Object {
      "condition": "new",
      "description": "Mayor rendimiento↵Su memoria RAM de 3 GB te permitirá ej..",
      "free_shipping": false,
      "id": "MLA846526085",
      "picture": "http://mla-s2-p.mlstatic.com/662389-MLA31003000984_062019-I.jpg",
      "price": Object {
        "amount": 139989.99,
        "currency": "$",
        "decimals": 0,
      },
      "sold_quaranty": 1,
      "title": "Apple iPhone 8 Plus 64 Gb Gris Espacial 3 Gb Ram",
    }
  }
/>
`;

exports[`ItemDetails should render without props 1`] = `<ItemDetails />`;
